### PR TITLE
Update DESCRIPTION

### LINKED ISSUE #139: Assess upcoming change of `emmeans` impact and adjust code accordingly if needed
Dear R package maintainer:

 

I’m writing to you because you are the maintainer of one or more R packages that is somewhere among the reverse dependencies of the emmeans package. (I have listed those packages below my signature).

 

I am making a change in the user experience for the emmeans() function. It has a convenience feature where the user may specify contrasts in addition to the desired means. This can be done either via a contr argument or by using a two-sided formula. The problem is that it can quickly become an inconvenience feature when there is more than one primary factor. For example, with 3 factors each having 3 levels, the specification pairwise ~ fac1*fac2*fac3 produces 378 pairwise comparisons, including a bunch of “diagonal” ones that the user likely does not want. This kind of thing creates all sorts of problems for unexperienced users, and unfortunately, it is the inexperienced users who are most likely to use this pairwise convenience feature as their primary “recipe.”

 

I had considered deprecating the contr argument and two-sided formulas. They are artifacts of an ancient version of lsmeans where that was the only way you could get contrasts, and they exist now only because of backward compatibility. But deprecation would create a huge amount of chaos in package incompatibilities and confused users. Plus, it really is a pretty convenient feature when you have just one primary factor,

 

First of all, I will say that the change that I am making should not affect any functionality or testing of your package. It only changes what happens to an interactive user who makes a direct call to emmeans(), and only then if there is more than one primary factor (as opposed to “by” factors). If all those are true, then a message and a menu appears giving the user an option of skipping the contrasts, and a little bit of guidance on what to do in case they do skip the contrasts. An example below illustrates this. Note that the suggestions make specific reference to the type of contrast desired (e.g., “pairwise”) and the existing factor names.

 

Again, this shouldn’t affect any package tests, because they are not interactive. And it won’t affect any use you make of emmeans() in your package because it only comes up with sys.parent() == 0, that is, a direct user call to emmeans(). But this could affect users of your package when they make such direct use of emmeans().

 

And of course I suggest reviewing your own package examples, and to try to encourage your users to get the means in one step, and any contrasts in subsequent steps.

 

Let me know if you have any comments or concerns.

 

Best,

 

--

Russ Lenth

 

Illustration of the new user experience:

```r

> warp.lm <- lm(breaks  wool * tension, data = warpbreaks)

> emmeans(warp.lm, pairwise ~ wool*tension)

 

You have asked for pairwise contrasts of wool*tension.

This might create a lot of output that perhaps you don't even want,

and you can get whatever contrasts you need later.

 

1: SKIP the contrasts (and see some instructions)

2: Go ahead and INCLUDE the contrasts

 

Selection: 1

 

wool tension emmean   SE df lower.CL upper.CL

A    L         44.6 3.65 48     37.2     51.9

B    L         28.2 3.65 48     20.9     35.6

A    M         24.0 3.65 48     16.7     31.3

B    M         28.8 3.65 48     21.4     36.1

A    H         24.6 3.65 48     17.2     31.9

B    H         18.8 3.65 48     11.4     26.1

 

Confidence level used: 0.95

 

Warning message:

Since your requested contrasts have been skipped,

try something like 'EMM <- .Last.value'

then run 'contrast(EMM, "pairwise", simple = "wool")'

See 'vignette("QuickStart", "emmeans")' for more information.
```

### PR DESCRIPTION
**Pull request by shajoezhu**
close #139

# Pull Request

Fixes #nnn

### CODE REVIEW

**Review by edelarua — COMMENTED** [review]
Lgtm!

**Comment by shajoezhu** [comment]
thanks Em! I am going to hold this off for some further testing. Should be all good. just checking

File: tests/testthat.R [hunk 1]
```diff
@@ -1,16 +1,13 @@
 pkg_name <- "tern.mmrm"
-if (requireNamespace("testthat", quietly = TRUE)) {
-  library(testthat)
-  library(vdiffr)
```
Comment (Melkiades) [inline tests/testthat.R:4]: makes sense to me!

**Review by Melkiades — APPROVED** [review]
Lgtm! Thanks Joe for simplifying `testthat.R` and threating vdiffr  as suggest

**Pull request merged**